cmake/developer_package/compile_flags/os_flags.cmake

```diff
@@ -256,12 +256,11 @@ if(WIN32)
     ie_add_compiler_flags(/Gy) # remove unreferenced functions: function level linking
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /LARGEADDRESSAWARE")
 
+    ie_add_compiler_flags(/WX)
```

Comment:
can we use CMAKE_COMPILE_WARNING_AS_ERROR in developer package?
```cmake
if(NOT DEFINED CMAKE_COMPILE_WARNING_AS_ERROR)
    set(CMAKE_COMPILE_WARNING_AS_ERROR ON)
endif()
```

Reply:
Do you propose to remove this compiler flag and always use cmake option?

Reply:
I suppose we can remove flags from this cmake file and set the mentioned option on CI (e.g. Azure Windows pipeline) explicitly. 
In this case always track the regressions, while people who want to compile using unsupported compiler which can emit new type of warnings, still can compile without errors.

Reply:
I propose to do it for the master branch only